targets/TARGET_STM/can_api.c

```diff
@@ -1102,6 +1102,10 @@ int can_mode(can_t *obj, CanMode mode)
 int can_filter(can_t *obj, uint32_t id, uint32_t mask, CANFormat format, int32_t handle)
 {
     int success = 0;
+
+    if(handle != 0) { // message filter handle not supported yet for STM controllers
+        return 0;
+    }
```

Comment:
I raised a separate issue ([#14596](https://github.com/ARMmbed/mbed-os/issues/14596)) relating to the CAN filter API, but as pointed out in [#11953](https://github.com/ARMmbed/mbed-os/issues/11953) the `handle` argument in `can_filter()` actually refers to a filter bank number, which is different from what the `handle` in `can_read()` is referring to. This edit in `can_filter()` would actually prevent filter banks other than bank 0 from being configured. Maybe a more appropriate solution is to rename `handle` to `filterBankNumber`?

Reply:
I think we can't change MBED API: https://github.com/ARMmbed/mbed-os/blob/master/hal/include/hal/can_api.h#L84 ?

Reply:
Oh I see, right. Although I still don't think this change in `can_filter()` is valid, because `handle` here isn't the same thing as `handle` in `can_read()`. This change will break the ability to set up multiple CAN filters.

Reply:
I've tested using multiple filters (and using dual CAN instances) on STM32F446RE and it was working?

Reply:
> Yes, multiple CAN filter seems not supported for STM32

The actual part of code where filter is initialized is at line number 1133, and init is done based on the CAN object passed. 
```
1133         if (HAL_CAN_ConfigFilter(&obj->CanHandle, &sFilterConfig) == HAL_OK)
1134         {
1135             success = 1;
1136         }
```

There is no race condition here instances on the platform. This will definitely not have any problems with any number of instances..

Reply:
What I'm trying to point out is that this part
```
if(handle != 0) { // message filter handle not supported yet for STM controllers
    return 0;
}
```
will exit the function if we pass in a nonzero `handle`. But I think that it should allow handle numbers 0-27 to be passed since there are 28 available filter banks.

From the [STM32F446xx reference manual](https://www.st.com/resource/en/reference_manual/dm00135183-stm32f446xx-advanced-arm-based-32-bit-mcus-stmicroelectronics.pdf):
![image](https://user-images.githubusercontent.com/36703188/119723690-b8572980-be3b-11eb-8d68-af0310e32fb2.png)

Reply:
Yes, the controller provides 0-27 filter banks, but Mbed does not have support for the filter handle for now.
This will be probably taken care in other PR.

Reply:
I tested on a STM32F446RE and I was able to set up multiple CAN filters by passing in different values for `handle`. I believe the functionality is there.